airbyte_cdk/sources/streams/http/requests_native_auth/abstract_oauth.py

```
@@ -217,10 +217,33 @@ def _make_handled_request(self) -> Any:
                 data=self.build_refresh_request_body(),
                 headers=self.build_refresh_request_headers(),
             )
+
```

nit: but wonder if there is a more straightforward way to write this logic. Something like

```
            # log the response even if the request failed for troubleshooting purposes
            self._log_response(response)
            response.raise_for_status()
            
            response_json = response.json()

            try:
                # extract the access token and add to secrets to avoid logging the raw value
                access_key = self._extract_access_token(response_json)
                if access_key is not None:
                  add_to_secrets(access_key)
            except ResponseKeysMaxRecurtionReached as e:
                # could not find the access token in the response, so do nothing
                pass
   
            return response_json
```

---

also, are we sure it makes sense to ignore the `ResponseKeysMaxRecurtionReached` exception? Admittedly this code is weird, but it seems like we could theoretically hit a max recursive depth but not fully explore the response json and therefore if we ignore the error, the access_key can actually be returned and we would not mask it.

---

Discussed this over zoom and came to a simpler implementation that we both liked